### PR TITLE
Issue #5722: Update DTD public names (typo at ImportControlLoader)

### PR DESCRIPTION
**Pull request by romani**
Issue #5722 

identified at https://github.com/checkstyle/checkstyle/issues/5601#issuecomment-396120018

CI failures https://circleci.com/gh/checkstyle/checkstyle/5997 :
```
4:14:57 AM PIT >> INFO : MINION : ls.checkstyle.api.CheckstyleException: unable to read
file:/home/circleci/project/src/test/resources/com/puppycrawl/tools/checkstyle/
checks/imports/importcontrol/InputImportControlOneRegExp.xml
at com.puppycrawl.tools.checkstyle.checks.imports.ImportContr
4:14:57 AM PIT >> INFO : MINION : olLoader.load(ImportControlLoader.java:282)
at com.puppycrawl.tools.checkstyle.checks.imports.ImportControlLoader.load(ImportControlLoader.java:253)
at com.puppycrawl.tools.checkstyle.checks.imports.ImportControlCheck.setFile(ImportControlCheck.java:192)
4:14:57 AM PIT >> INFO : MINION : 
... 44 more
Caused by: java.io.IOException: Server returned HTTP response code: 503 for URL:
http://checkstyle.sourceforge.net/dtds/import_control_1_2.dtd
at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1876)
```

the same problem detected Travis https://travis-ci.org/checkstyle/checkstyle/jobs/390563236#L446

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportControlLoader.java [hunk 1]
```diff
@@ -64,7 +64,7 @@
 
     /** The new public ID for version 1_2 of the configuration dtd. */
     private static final String DTD_PUBLIC_CS_ID_1_2 =
-        "-//Checkstyle//DTD ImportControl Configuration 1.22//EN";
+        "-//Checkstyle//DTD ImportControl Configuration 1.2//EN";
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportControlLoader.java]: Why wasn't this detected earlier?

**Comment by romani** [comment]
Public name allow loader to reuse DTD from jar.
If public name is not known, loader us going to web to get dtd. Web link was correct and downloadable.  Problem appears when SF site gone down on maintenance, so loader had no way to get dtd at all.

**Comment by rnveach** [comment]
> loader is going to web to get dtd

Shouldn't we disable this behavior?

We put all our DTDs to loader. There shouldn't be a reason to go to the web for anything we own. If user creates their own custom config, they need to create their own loader and can't pass it through ours. There has been a case before where we forgot to add new DTD to the loader and it wasn't found because of this.

**Comment by romani** [comment]
all external (out of jar) xml files  does not belong to checkstyle.
It is user preference what location schema to use. Such XML file should be editable and validated by external tools. External XML edit/validate tools  ignore Schema public ID, and use always URL (System ID). 
In our case we try to use public ID to reuse embedded DTD, we define System ID to be friendly to external world (all do the same, we are not uniq in our behavior).
But when both defines, standard/jdk  XML loader try to use Public and then System ID. 
One on them always worked :), but not when SF went down for maintenance.
As gt typo in publicID name , our UT become network/Internet dependent, non of us usually are offline to catch it quickly.

We can probably check it by some sort of disabling System ID usage, but I do not know how to make it.
If you are interested, you can open issue on this and investigate.

**Comment by rnveach** [comment]
> It is user preference what location schema to use

Location is ignored completely. We could make it some bad url and CLI would still work because we override based on the public ID. It is only useful to tools outside checkstyle who don't know about us.

> probably check it by some sort of disabling System ID usage

There is no way to pick up an internet connection unless we implement a proxy, and you have said in another issue you are against this.
The best way is to go forward with this is to instantly fail when it is a Public ID we don't know. It won't impact us at all.

**Comment by rnveach** [comment]
Here is example:
````
$ cat TestClass.java
public class TestClass {
    void method() {
    }
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://IwishIcouldRememberThisLongNameButWhoCaresAnyways/dtds/WhatAm_I_Named_9999_z.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
    </module>
</module>

$ java -jar checkstyle-8.10.1-all.jar -c TestConfig.xml TestClass.java
Starting audit...
Audit done.
````

**Comment by romani** [comment]
> It is only useful to tools outside checkstyle who don't know about us and our DTD.

please remove system ID and validate user XML config by some tool. Please share on how to make it.

> The best way is to go forward with this is to instantly fail when it is a Public ID we don't know. It won't impact us at all.

ones we(our cli) process files we can do this, I am not sure it is good approach.
I do not think we need to invest time in this, it is better to spend this time on switch from DTD to XSD. DTD is really ancient schema definition, less and less tools consider it for support (as we had a problem with xml-maven-plugin  recently).

**Pull request merged**